### PR TITLE
Add inference api docs

### PR DESCRIPTION
**Pull request by andrewm4894**
- add inference api docs to docs site.
- add instructions for manually updating them.
- add title to inference server fast api app so it shows up properly in the docs.

Ends up looking like below where docusaurus just handles the multiple *-openapi.json pretty well

image

### CODE REVIEW

**Review by olliestanley — APPROVED** [review]
This will be really great especially for new contributors working with inference, thank you!

**Comment by andrewm4894** [comment]
Planning on making another go at trying to automate the update of the api docs (via the api contract tests workflow) too. Will try get to that next.

File: inference/server/oasst_inference_server/settings.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ def split_keys_string(keys: str | None):
 
 
 class Settings(pydantic.BaseSettings):
+    PROJECT_NAME: str = "open-assistant inference server"
```
Comment (andreaskoepf) [inline inference/server/oasst_inference_server/settings.py:13]: all other settings are lower-snake-case .. maybe we should be consistent here and make it `project_name`?

**Review by melvinebenezer — APPROVED** [review]
apart from the lower-snake-case comment. LGTM

File: inference/server/oasst_inference_server/settings.py (hunk 1 above)
Comment (andrewm4894) [reply inference/server/oasst_inference_server/settings.py:13]: Yeah, was just copying the equivalent in the backend code as close as possible that's only reason I did this way. No strong feelings either way.

**Pull request merged**